Tools/ML/ONNX/model.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file     model.h
+///
+/// \author   Christian Sonnabend <christian.sonnabend@cern.ch>
+///
+/// \brief    A general-purpose class for ONNX models
+///
+
+#ifndef TOOLS_ML_ONNX_MODEL_H_
+#define TOOLS_ML_ONNX_MODEL_H_
+
+// C++ and system includes
+#include <onnxruntime/core/session/experimental_onnxruntime_cxx_api.h>
+#include <vector>
+#include <string>
+#include <memory>
+#include <map>
+
+// ROOT includes
+#include "TSystem.h"
+
+// O2 includes
+#include "CCDB/CcdbApi.h"
+#include "Framework/Logger.h"
+
+namespace o2
+{
+
+namespace ml
+{
+
+class OnnxModel
+{
+
+ public:
+  OnnxModel();
+  explicit OnnxModel(std::string);
+  ~OnnxModel() = default;
+  OnnxModel& operator=(OnnxModel&);
+
+  // IO
+  std::map<std::string, std::string> fetchHeaders(std::string, int64_t, bool);
+  bool downloadToFile(std::string, int64_t, std::string, bool);
+  bool fetchFromCCDB(std::string, int64_t, std::string, bool);
```

Comment:
Do these functions need to be exposed to the user? In general, I would keep the interface as simple as possible.

Reply:
Not necessarily, but sometimes a user might want to download a model without loading it into the class immediately, that's why I added these two functions. But it depends on what the goal for the class would be.

Reply:
I would try to enforce a one-to-one relation between an ML model and an instance of this class. In which cases do you think it would be helpful to separate the fetching of the model from creating the class object?